battle/boss.py

```diff
@@ -402,9 +402,14 @@ def geneaux():
         battle.main.buddy_swap(Kimahri)
         CurrentPlayer().attack()
         while not Tidus.is_turn():
-            CurrentPlayer().defend()
+            if memory.main.turn_ready():
```

Comment:
I feel like we should roll this check into `Player.is_turn()`. Can we think of an instance where we want to do things before their turn is ready? I don't think so. So moving this checking into that method would likely fix a bunch of timing stuff.

Reply:
hmmm... there are times when we'd want to check specifically turn_ready without tagging it to any player (like waiting for the first/next turn), but yeah I think we could do that eventually.